src/cls/IPM/Utils/Module.cls

```diff
@@ -1172,6 +1434,15 @@ ClassMethod LoadNewModule(
         }
         $$$ThrowOnError(tSC)
         do tModule.WriteAfterInstallMessage()
+
+        // After install of modules and dependencies is completed, create the lock file for the module if flag specified
+        if $get(params("CreateLockFile"), 0) {
```

Comment:
Move this to wherever the dependency graph is constructed so that can be passed here